android/src/main/java/com/robinpowered/RNMDMManager/RNMobileDeviceManagerModule.java

```diff
@@ -80,7 +137,47 @@ public void getConfiguration(final Promise promise) {
             }
             promise.resolve(data);
         } else {
-          promise.reject(new Error("Managed App Config is not supported"));
+            promise.reject(new Error("Managed App Config is not supported"));
+        }
+    }
+
+    @ReactMethod
+    public void isAutonomousSingleAppModeSupported(final Promise promise) {
+        promise.resolve(isLockStatePermitted());
+    }
+
+    @ReactMethod
+    public void isSingleAppModeEnabled(final Promise promise) {
+        promise.resolve(isLockState());
+    }
+
+    @ReactMethod
+    public void isAutonomousSingleAppModeEnabled(final Promise promise) {
+        boolean lockedStatus = false;
+        ActivityManager am = (ActivityManager) getReactApplicationContext().getSystemService(Context.ACTIVITY_SERVICE);
+        try {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                if (am.getLockTaskModeState() == ActivityManager.LOCK_TASK_MODE_LOCKED) {
+                    lockedStatus = true;
+                }
+            } else {
+                if (am.isInLockTaskMode() && isLockStatePermitted()) {
```

Comment:
`lockedStatus = am.isInLockTaskMode() && isLockStatePermitted()`

Note - should probably do the permitted check first since we wouldn't need to check it if it's not permitted, right?